polygeist/tools/cgeist/Options.h

```diff
@@ -13,9 +13,33 @@
 #ifndef CGEIST_OPTIONS_H_
 #define CGEIST_OPTIONS_H_
 
+#include "llvm/ADT/Optional.h"
+#include "llvm/Passes/OptimizationLevel.h"
 #include "llvm/Support/CommandLine.h"
 #include <string>
 
+/// Class wrapping cgeist options.
+class CgeistOptions {
+public:
+  CgeistOptions(bool SYCLKernelsOnly,
+                llvm::Optional<llvm::OptimizationLevel> OptimizationLevel)
+      : SYCLKernelsOnly(SYCLKernelsOnly), OptimizationLevel(OptimizationLevel) {
+  }
+
+  constexpr bool getSYCLKernelsOnly() const { return SYCLKernelsOnly; }
+
+  constexpr const llvm::Optional<llvm::OptimizationLevel> &
+  getOptimizationLevel() const {
+    return OptimizationLevel;
+  }
+
+private:
+  /// Whether to generate MLIR only for kernels.
+  const bool SYCLKernelsOnly = false;
```

Comment:
`SYCLKernelsOnly` is true when given `-fsycl-is-device`, is `SYCLDeviceOnly` a better name for that?